docs/sbom-tool-api-reference.md

```diff
@@ -0,0 +1,192 @@
+
+```C#
+using Microsoft.Sbom.Contracts;
+
+var spdx22Specification = new SBOMSpecification("SPDX", "2.2");
+```
+
+Our tool is designed to support multiple formats of SBOM, however it currently only supports SPDX v2.2. If you have additional SBOM formats that are implememnted, this handy API call can provide you with a list of all the formats that are currently supported by our tool.
+
+```C#
+using Xunit;
+
+var specifications = generator.GetSupportedSBOMSpecifications();
+
+Assert.True(specifications.Count() == 1);
+Assert.Equal("SPDX", specifications.First().Name);
+Assert.Equal("2.2", specifications.First().Version);
+```
+
+### GetRequiredAlgorithms
+
+Each SBOM specification has a list of hash algorithms that are required to be generated for each package and file. This handy API will provide a list of hashing algorithms that are required for a specific SBOM format.
+
+```C#
+
+var algorithms = generator.GetRequiredAlgorithms(spdx22Specification);
+
+Assert.True(algorithms.Count() == 2);
+Assert.True(algorithms.Any(a => a.Name == "SHA256"));
+Assert.True(algorithms.Any(a => a.Name == "SHA1"));
+
+```
+
+This API is helpful when using the self provided data based API, as the caller would have to provide these hashes for all the packages and files they send as part of the input data.
+
+## Prerequisites for calling the API
+
+In order to call the API, you must first generate one required and one optional data object.
+
+### SBOMMetadata
+
+The `SBOMMetadata` object provides the SBOM tool with additional metadata that can be used to configure some output metadata values in the SBOM, for example the product name or version.
+
+```C#
+
+SBOMMetadata metadata = new SBOMMetadata()
+{
+    PackageName = "MyProject", //Required
+    PackageVersion = "0.0.1", // Required
+    BuildId = "2344", // Optional
+    BuildEnvironmentName = "Github Actions" // Optional
+};
+```
+
+The metadata object **must** be created and passed to the API as it contains two required values for the SBOM generator to run, the package name and version. These values are used in the generated SBOM to define the name and documentNamespace. The other keys in the metadata object are optional, and the tool may or may not use these values to generate additional metadata in the created SBOM.
+
+
+### RuntimeConfiguration
+
+The `RuntimeConfiguration` object contains configuration that affects the actual execution of the SBOM tool. It contains configurations like Verbosity that can affect how much logging is returned by our tool. 
+
+```C#
+RuntimeConfiguration configuration = new RuntimeConfiguration()
+{
+    DeleteManifestDirectoryIfPresent = true,
+    WorkflowParallelism = 8,
+    Verbosity = System.Diagnostics.Tracing.EventLevel.Verbose,
+    NamespaceUriBase = "http://sbom.mycompany.com"
```

Comment:
Not relevant to this PR, but is actually feels a little bit off that NamespaceUriBase is in RuntimeConfig rather than SBOMMetadata object

Reply:
Yeah, that's what I thought while writing this doc. I will open a new PR and deprecate it here and move it to the metadata object.